kernel/src/engine/arrow_expression/evaluate_expression.rs

```diff
@@ -132,9 +132,18 @@ fn evaluate_transform_expression(
     // Collect output columns directly to avoid creating intermediate Expr::Column instances.
     let mut output_cols = Vec::new();
 
+    // Helper lambda to get the next output field type
+    let mut output_schema_iter = output_schema.fields();
+    let mut next_output_type = || {
+        output_schema_iter
+            .next()
+            .map(|field| field.data_type())
+            .ok_or_else(|| Error::generic("Output schema mismatch"))
```

Comment:
Should we maybe make the error more specific? IIUC, this should be something like `Output schema mismatch - expected more columns`.